src/backend/base/langflow/components/agents/ToolCallingAgent.py

```diff
@@ -43,4 +43,14 @@ def create_agent_runnable(self):
             ("placeholder", "{agent_scratchpad}"),
         ]
         prompt = ChatPromptTemplate.from_messages(messages)
-        return create_tool_calling_agent(self.llm, self.tools, prompt)
+        try:
+            return create_tool_calling_agent(self.llm, self.tools, prompt)
+        except NotImplementedError as e:
+            import langchain
+
+            message = (
+                f"The model '{self.llm.__class__.__name__}' is not supported "
+                f"in version '{langchain.__version__}' of Langchain. "
+                "Please try using a different, compatible model."
+            )
+            raise NotImplementedError(message) from e
```

Comment:
I think something like this would be best
```suggestion
        try:
            return create_tool_calling_agent(self.llm, self.tools, prompt)
        except NotImplementedError as e:
            message = (
                f"{self.display_name} does not support tool calling."
                "Please try using a compatible model."
            )
            raise NotImplementedError(message) from e
```